### ISSUE TITLE
batch_size问题,backbone为resnest101,在4块titan X上可以开到64，但是在4块2080ti上仅能开到8.

### ISSUE THREAD

**Issue by yanty123** [JDAI-CV/fast-reid#364]
类别数为127817，模型确实很大，但是在titan/2080ti显存差不多的情况下（12G左右）上batch却差了太多（64 vs 8），希望大大能帮忙解决~

**下面是在titan X上的config：**

`CUDNN_BENCHMARK: true
DATALOADER:
  NAIVE_WAY: true
  NUM_INSTANCE: 4
  NUM_WORKERS: 8
  PK_SAMPLER: true
  PLATE_ERASE: false
DATASETS:
  COMBINEALL: false
  NAMES:
  - Alidata
  TESTS:
  - VeRi
INPUT:
  CJ:
    BRIGHTNESS: 0.35
    CONTRAST: 0.35
    ENABLED: true
    HUE: 0.2
    PROB: 0.8
    SATURATION: 0.35
  DO_AUGMIX: false
  DO_AUTOAUG: false
  DO_FLIP: true
  DO_PAD: true
  FLIP_PROB: 0.5
  PADDING: 10
  PADDING_MODE: constant
  REA:
    ENABLED: true
    MEAN:
    - 123.675
    - 116.28
    - 103.53
    PROB: 0.5
  RPT:
    ENABLED: false
    PROB: 0.5
  SIZE_TEST:
  - 224
  - 224
  SIZE_TRAIN:
  - 224
  - 224
MODEL:
  BACKBONE:
    DEPTH: 101x
    FEAT_DIM: 2048
    LAST_STRIDE: 1
    NAME: build_resnest_backbone
    NORM: BN
    PRETRAIN: true
    PRETRAIN_PATH: ''
    WITH_IBN: true
    WITH_NL: false
    WITH_SE: false
  DEVICE: cuda
  FREEZE_LAYERS:
  - ''
  HEADS:
    CLS_LAYER: linear
    EMBEDDING_DIM: 0
    MARGIN: 0.15
    NAME: EmbeddingHead
    NECK_FEAT: before
    NORM: BN
    NUM_CLASSES: 0
    POOL_LAYER: avgpool
    SCALE: 128
    WITH_BNNECK: true
  LOSSES:
    CE:
      ALPHA: 0.2
      EPSILON: 0.1
      SCALE: 0.5
    CIRCLE:
      ALPHA: 128
      MARGIN: 0.25
      SCALE: 1.0
    FL:
      ALPHA: 0.25
      GAMMA: 2
      SCALE: 1.0
    NAME:
    - CrossEntropyLoss
    - TripletLoss
    TRI:
      HARD_MINING: true
      MARGIN: 0.3
      NORM_FEAT: false
      SCALE: 0.5
  META_ARCHITECTURE: Baseline
  PIXEL_MEAN:
  - 123.675
  - 116.28
  - 103.53
  PIXEL_STD:
  - 58.395
  - 57.120000000000005
  - 57.375
  WEIGHTS: ''
OUTPUT_DIR: /home/yantianyi/logs/resnest101_ali
SOLVER:
  AMP_ENABLED: false
  BASE_LR: 0.00035
  BIAS_LR_FACTOR: 2.0
  CHECKPOINT_PERIOD: 10
  DELAY_ITERS: 0
  ETA_MIN_LR: 3.0e-07
  FREEZE_ITERS: 0
  GAMMA: 0.1
  HEADS_LR_FACTOR: 1.0
  IMS_PER_BATCH: 64
  MAX_ITER: 90
  MOMENTUM: 0.9
  OPT: Adam
  SCHED: WarmupMultiStepLR
  STEPS:
  - 25
  - 40
  SWA:
    ENABLED: false
    ETA_MIN_LR: 3.5e-06
    ITER: 10
    LR_FACTOR: 10.0
    LR_SCHED: false
    PERIOD: 2
  WARMUP_FACTOR: 0.01
  WARMUP_ITERS: 10
  WARMUP_METHOD: linear
  WEIGHT_DECAY: 0.0005
  WEIGHT_DECAY_BIAS: 0.0005
TEST:
  AQE:
    ALPHA: 3.0
    ENABLED: false
    QE_K: 5
    QE_TIME: 1
  EVAL_PERIOD: 30
  IMS_PER_BATCH: 128
  METRIC: cosine
  PRECISE_BN:
    DATASET: Market1501
    ENABLED: false
    NUM_ITER: 300
  RERANK:
    ENABLED: false
    K1: 20
    K2: 6
    LAMBDA: 0.3
  ROC_ENABLED: false
`
**下面是在2080ti上的config：**

`CUDNN_BENCHMARK: true
DATALOADER:
  NAIVE_WAY: true
  NUM_INSTANCE: 4
  NUM_WORKERS: 8
  PK_SAMPLER: true
  PLATE_ERASE: false
DATASETS:
  COMBINEALL: false
  NAMES:
  - Alidata
  TESTS:
  - VeRi
INPUT:
  CJ:
    BRIGHTNESS: 0.35
    CONTRAST: 0.35
    ENABLED: true
    HUE: 0.2
    PROB: 0.8
    SATURATION: 0.35
  DO_AUGMIX: false
  DO_AUTOAUG: false
  DO_FLIP: true
  DO_PAD: true
  FLIP_PROB: 0.5
  PADDING: 10
  PADDING_MODE: constant
  REA:
    ENABLED: true
    MEAN:
    - 123.675
    - 116.28
    - 103.53
    PROB: 0.5
  RPT:
    ENABLED: false
    PROB: 0.5
  SIZE_TEST:
  - 224
  - 224
  SIZE_TRAIN:
  - 224
  - 224
MODEL:
  BACKBONE:
    DEPTH: 101x
    FEAT_DIM: 2048
    LAST_STRIDE: 1
    NAME: build_resnest_backbone
    NORM: BN
    PRETRAIN: true
    PRETRAIN_PATH: ''
    WITH_IBN: true
    WITH_NL: false
    WITH_SE: false
  DEVICE: cuda
  FREEZE_LAYERS:
  - ''
  HEADS:
    CLS_LAYER: linear
    EMBEDDING_DIM: 0
    MARGIN: 0.15
    NAME: EmbeddingHead
    NECK_FEAT: before
    NORM: BN
    NUM_CLASSES: 0
    POOL_LAYER: avgpool
    SCALE: 128
    WITH_BNNECK: true
  LOSSES:
    CE:
      ALPHA: 0.2
      EPSILON: 0.1
      SCALE: 0.5
    CIRCLE:
      ALPHA: 128
      MARGIN: 0.25
      SCALE: 1.0
    FL:
      ALPHA: 0.25
      GAMMA: 2
      SCALE: 1.0
    NAME:
    - CrossEntropyLoss
    - TripletLoss
    TRI:
      HARD_MINING: true
      MARGIN: 0.3
      NORM_FEAT: false
      SCALE: 0.5
  META_ARCHITECTURE: Baseline
  PIXEL_MEAN:
  - 123.675
  - 116.28
  - 103.53
  PIXEL_STD:
  - 58.395
  - 57.120000000000005
  - 57.375
  WEIGHTS: ''
OUTPUT_DIR: /home/yantianyi/logs/resnest101_ali
SOLVER:
  AMP_ENABLED: false
  BASE_LR: 0.00035
  BIAS_LR_FACTOR: 2.0
  CHECKPOINT_PERIOD: 10
  DELAY_ITERS: 0
  ETA_MIN_LR: 3.0e-07
  FREEZE_ITERS: 0
  GAMMA: 0.1
  HEADS_LR_FACTOR: 1.0
  IMS_PER_BATCH: 8
  MAX_ITER: 90
  MOMENTUM: 0.9
  OPT: Adam
  SCHED: WarmupMultiStepLR
  STEPS:
  - 25
  - 40
  SWA:
    ENABLED: false
    ETA_MIN_LR: 3.5e-06
    ITER: 10
    LR_FACTOR: 10.0
    LR_SCHED: false
    PERIOD: 2
  WARMUP_FACTOR: 0.01
  WARMUP_ITERS: 10
  WARMUP_METHOD: linear
  WEIGHT_DECAY: 0.0005
  WEIGHT_DECAY_BIAS: 0.0005
TEST:
  AQE:
    ALPHA: 3.0
    ENABLED: false
    QE_K: 5
    QE_TIME: 1
  EVAL_PERIOD: 30
  IMS_PER_BATCH: 128
  METRIC: cosine
  PRECISE_BN:
    DATASET: Market1501
    ENABLED: false
    NUM_ITER: 300
  RERANK:
    ENABLED: false
    K1: 20
    K2: 6
    LAMBDA: 0.3
  ROC_ENABLED: false
`

**Comment by L1aoXingyu** [comment]
使用新版 code，把 fp_16 混合精度训练打开

**Comment by yanty123** [comment]
> 使用新版 code，把 fp_16 混合精度训练打开

感谢！ 是 SOLVER.AMP_ENABLED:True 对吧？
但是还是一样，batch开不大。而且我很奇怪为什么同样的代码和设置在titan x上就没有问题？

**Comment by L1aoXingyu** [comment]
新版 code 里面改成了 `fp16_enabled`，是不是你的 pytorch 版本和 cuda 版本不对

**Comment by yanty123** [comment]
> 新版 code 里面改成了 `fp16_enabled`，是不是你的 pytorch 版本和 cuda 版本不对，titanX 的显存和 2080ti 的显存差距大吗？

好的 谢谢您 我再试试， 显存差距不大的，都是12G左右。
